src/synthesis.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  availableAmount,
+  itemAmount,
+  mySpleenUse,
+  spleenLimit,
+  sweetSynthesis,
+  sweetSynthesisResult,
+} from "kolmafia";
+import { $items, clamp } from "libram";
+import { shuffle } from "./lib";
+
+const whitelist = $items`sugar shotgun, sugar shillelagh, sugar shank, sugar chapeau, sugar shorts, sugar shield, sugar shirt, Fudgie Roll`;
+export default function synthesize(effect: Effect, casts: number): void {
+  const shuffledWhitelist = shuffle([...whitelist]);
+  for (const itemA of shuffledWhitelist) {
+    if (availableAmount(itemA) <= 1) continue;
```

Comment:
```suggestion
    if (itemAmount(itemA) <= 1) continue;
```

Reply:
My concern with these changes is that it won't pull candies from hagnks--wouldn't we rather use availableAmount and use retrieveItem, rather than using itemAmount to begin with?